packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -800,6 +801,20 @@ export class Project extends ProjectBase {
               account: parsedArn?.account,
               region: parsedArn?.region,
             }));
+            // if secret comes from another account, SecretsManager will need to access
+            // KMS on the other account as well to be able to get the secret
+            if (parsedArn?.account !== stack.account) {
+              kmsIamResources.push(stack.formatArn({
+                service: 'kms',
+                resource: 'key',
+                resourceName: '*',
```

Comment:
A comment here saying something like "We don't know the ID of the Key, but since this is a cross-account access, the Key will have to define the correct trust policy to allow this access anyway, so a `"*"` here is safe" would be nice 🙂.